### PR TITLE
Add audio file playback capabilities using TTS Chunks

### LINKED ISSUE #1277: [SDL 0014] Adding Audio File Playback to TTSChunk
Proposal: [Adding Audio File Playback to TTSChunk](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0014-adding-audio-file-playback-to-ttschunk.md)

> More and more applications use sounds as a means to make a user feel more at home within their application, and allow the developer to have more customization of their User Experience. This proposal will allow developers to provide audio files uploaded to be played in conjunction with, or without, Text-to-Speech text, so that they may provide a more customized experience.

Review: smartdevicelink/sdl_evolution#42

Steering Committee Decision:

> This proposal was accepted. The committee feels it is a good feature to add to give apps some personality and to enable new features, and while the naming isn't ideal, it gives us the maximum amount of flexibility with the minimum extra development cost.  The implementation will have to be fleshed out on core to ensure backward compatibility.

### PR DESCRIPTION
**Pull request by jacobkeeler**
SDL Core implementation of [SDL-0014](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0014-adding-audio-file-playback-to-ttschunk.md)

Resolves #1277 

Documentation PR: smartdevicelink/sdl_hmi_integration_guidelines#58
ATF test updates: smartdevicelink/sdl_atf_test_scripts#1257

### CODE REVIEW

**Comment by AByzhynar** [comment]
Rebuild required

**Comment by AByzhynar** [comment]
Rebuild required

File: src/components/interfaces/MOBILE_API.xml [hunk 1]
```diff
@@ -136,7 +136,7 @@
             <description>The value being set is read only</description>
         </element>
     </enum>
-    
+
```
Comment (askirk) [inline src/components/interfaces/MOBILE_API.xml]: All of these types of changes in this file are unneccessary and don't relate to the proposal. Instead, only the one line related to adding the FILE enum should appear.

File: src/components/application_manager/test/commands/mobile/slider_test.cc [hunk 2]
```diff
@@ -282,9 +282,8 @@ TEST_F(SliderRequestTest, OnEvent_UI_OnResetTimeout_UNSUCCESS) {
   CommandPtr command(CreateCommand<SliderRequest>(msg_));
   EXPECT_TRUE(command->Init());
 
-  EXPECT_CALL(
-      app_mngr_,
-      updateRequestTimeout(kConnectionKey, kCorrelationId, _));
+  EXPECT_CALL(app_mngr_,
+              updateRequestTimeout(kConnectionKey, kCorrelationId, _));
```
Comment (askirk) [inline src/components/application_manager/test/commands/mobile/slider_test.cc]: This change is unrelated to the proposal, should remove

**Comment by AStasiuk** [comment]
@askirk could you please tag @theresalech to this PR? This will show that this PR is ready for Livio review.

**Comment by JackLivio** [comment]
@jacobkeeler 

I have a couple of comments.

I believe there should be some sort of error checking by sdl core to make sure the audio file exists before passing the file name to the hmi. I assume the audio chunks uploaded by the app would be stored in the app's storage directory because they are received by a put file?

If the file does not exist, at the very least Core should respond with a warning to the mobile app.

Another note is that most file names that are passed to the HMI include the full file path. On the HMI API side for TTSChunk, I believe that the `text` parameter should contain the full file path of the audio chunk file.
